### PR TITLE
Clarify ParquetExample inputs and outputs

### PR DESCRIPTION
**Pull request by clairemcginty**
when I was running the `ParquetExample`s in Dataflow I noticed that the `--typedOut` job doesn't create a valid input for `--typedIn`. I cleaned up the input/output types and added some clarifying comments so that the `--{avroSpecific|avroGeneric|typed}In` jobs have valid dependencies on the `--{avro|typed}Out ` jobs.

### CODE REVIEW

File: scio-examples/src/test/scala/com/spotify/scio/examples/extra/ParquetExampleTest.scala [hunk 1]
```diff
@@ -18,74 +18,57 @@ package com.spotify.scio.examples.extra
...
-      .input(ParquetTypeIO[AccountInput]("in.parquet"), input)
+      .input(ParquetTypeIO[AccountProjection]("in.parquet"), input)
       .output(TextIO("out.txt"))(coll => coll should containInAnyOrder(expected))
       .run()
   }
 
-  "ParquetExample" should "work for specific output" in {
-    val expected = dummyData
-
+  it should "work for specific output" in {
```
Comment (stormy-ua) [inline scio-examples/src/test/scala/com/spotify/scio/examples/extra/ParquetExampleTest.scala:54]: maybe reword it to "work for Avro specific output"?

**Review by stormy-ua — APPROVED** [review]

File: scio-examples/src/test/scala/com/spotify/scio/examples/extra/ParquetExampleTest.scala (hunk 1 above)
Comment (clairemcginty) [reply scio-examples/src/test/scala/com/spotify/scio/examples/extra/ParquetExampleTest.scala:54]: good idea! I will 👍
